test/unit_tests/autoqasm/test_api.py

```diff
@@ -1266,3 +1266,43 @@ def test(int[32] a, int[32] b) {
 test(2, 3);
 test(4, 5);"""
     assert main.build().to_ir() == expected
+
+
+class TestTypecasting:
+    def test_int_typecasting_on_measure(self):
+        @aq.main(num_qubits=2)
+        def main():
+            test = int(measure([0, 1]))  # noqa: F841
+
+        expected_ir = """OPENQASM 3.0;
+qubit[2] __qubits__;
+bit[2] __bit_0__ = "00";
+__bit_0__[0] = measure __qubits__[0];
+__bit_0__[1] = measure __qubits__[1];
+int[32] test = __bit_0__;"""
+        assert main.build().to_ir() == expected_ir
+
+    def test_int_typecasting_on_string(self):
+        @aq.main(num_qubits=2)
+        def main():
+            test = int("101", 2)  # noqa: F841
+
+        expected_ir = """OPENQASM 3.0;
+qubit[2] __qubits__;"""
+        assert main.build().to_ir() == expected_ir
+
+    def test_nested_int_typecasting(self):
+        @aq.main(num_qubits=2)
+        def main():
+            test = 2 * int(measure([0, 1]))  # noqa: F841
+            return test
```

Comment:
Added! Could you confirm that in the case where there is no return statement, there should not be an instruction involving the ` 2 *`. Currently, I am getting:

*with* return
```
OPENQASM 3.0;
output int[32] test;
qubit[2] __qubits__;
bit[2] __bit_0__ = "00";
__bit_0__[0] = measure __qubits__[0];
__bit_0__[1] = measure __qubits__[1];
int[32] __int_1__;
__int_1__ = __bit_0__;
test = 2 * __int_1__;
```

*without* return
```
OPENQASM 3.0;
qubit[2] __qubits__;
bit[2] __bit_0__ = "00";
__bit_0__[0] = measure __qubits__[0];
__bit_0__[1] = measure __qubits__[1];
int[32] __int_1__;
__int_1__ = __bit_0__;
```

Reply:
That's interesting - I still would expect the `test` variable to be declared and assigned `2 * __int_1__` there. But functionally it doesn't matter, of course. I expect this would be an issue in the assignment converter and outside the scope of your PR. It shouldn't block moving forward with your changes.

Reply:
Maybe I can look into this in a new PR?